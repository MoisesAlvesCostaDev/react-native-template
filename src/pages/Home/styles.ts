import styled from 'styled-components/native';
import {BACKGROUND_COLOR_PRIMARY, BACKGROUND_COLOR_SECONDARY} from "assets/colors";


export const Screen = styled.View`
  flex: 1;
  background-color: ${BACKGROUND_COLOR_PRIMARY};
`;


export const HeaderContainer = styled.View`
  flex: 1;
`;


export const BodyContainer = styled.View`
  flex: 11;
  background-color: ${BACKGROUND_COLOR_SECONDARY};
  border-top-left-radius: 30px;
  border-top-right-radius: 30px;
  padding-top:20px;

`;
