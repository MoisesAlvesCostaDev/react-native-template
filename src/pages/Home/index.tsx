import React from "react";
import { Screen } from "assets/styles/Screen";
import { BodyContainer, HeaderContainer } from "./styles";

const Home = function Home(): JSX.Element {
  return (
    <Screen>
      <HeaderContainer></HeaderContainer>
      <BodyContainer></BodyContainer>
    </Screen>
  );
};

export default Home;
