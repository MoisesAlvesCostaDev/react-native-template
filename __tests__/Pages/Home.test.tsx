import "react-native";
import React from "react";
import Home from "pages/Home";
import { render } from "@testing-library/react-native";

describe("test render Login screen", () => {
  it("renders correctly", () => {
    const { getByText, toJSON } = render(<Home />);
    getByText("Log out");
    expect(toJSON()).toMatchSnapshot();
  });
});
