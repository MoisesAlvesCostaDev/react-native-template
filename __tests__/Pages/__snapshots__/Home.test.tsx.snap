// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Login screen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FF5C00",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderTopLeftRadius": 30,
          "borderTopRightRadius": 30,
          "flexBasis": 0,
          "flexGrow": 11,
          "flexShrink": 1,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#EDEDED",
          "borderRadius": 12,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID="buttonStyled"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#646464",
              "fontSize": 16,
            },
          ]
        }
      >
        Log out
      </Text>
    </View>
  </View>
</View>
`;
